crates/pallets/starknet/src/lib.rs

```diff
@@ -1166,6 +1124,98 @@ impl<T: Config> Pallet<T> {
         Ok(results)
     }
 
+    pub fn simulate_transactions(
+        transactions: Vec<UserTransaction>,
+        simulation_flags: Vec<SimulationFlag>,
+    ) -> Result<Vec<SimulatedTransaction>, DispatchError> {
+        let chain_id = Self::chain_id();
+
+        let mut execution_config_builder = RuntimeExecutionConfigBuilder::default();
+        if simulation_flags.contains(&SimulationFlag::SkipFeeCharge) {
+            execution_config_builder = execution_config_builder.with_fee_charge_disabled();
+        }
+        if simulation_flags.contains(&SimulationFlag::SkipValidate) {
```

Comment:
Looks like you can an `with_simulation_flags()` method to `RuntimeExecutionConfigBuilder` and hide all this